lib/ecto/enum.ex

```diff
@@ -46,17 +50,25 @@ defmodule Ecto.Enum do
   def init(opts) do
     values = Keyword.get(opts, :values, nil)
 
-    unless is_list(values) and Enum.all?(values, &is_atom/1) do
+    unless (is_list(values) and
+              Enum.all?(values, &is_atom/1)) or
+             (Keyword.keyword?(values) and Enum.all?(Keyword.values(values), &is_integer/1)) do
       raise ArgumentError, """
       Ecto.Enum types must have a values option specified as a list of atoms. For example:
 
           field :my_field, Ecto.Enum, values: [:foo, :bar]
       """
     end
 
-    on_load = Map.new(values, &{Atom.to_string(&1), &1})
-    on_dump = Map.new(values, &{&1, Atom.to_string(&1)})
-    %{on_load: on_load, on_dump: on_dump, values: values}
+    if Enum.all?(values, &is_atom/1) do
+      on_load = Map.new(values, &{Atom.to_string(&1), &1})
+      on_dump = Map.new(values, &{&1, Atom.to_string(&1)})
+      %{on_load: on_load, on_dump: on_dump, values: values}
+    else
+      on_load = Map.new(values, fn {key, val} -> {val, key} end)
+      on_dump = Enum.into(values, %{})
+      %{on_load: on_load, on_dump: on_dump, values: values}
```

Comment:
Please normalize `values` to always be a map so we don't need to check for `Keyword.keyword?` later on.

Reply:
That's a good point. I have changed it in 8f0b227f5db0eb7347c4601cac4f8377b13abd66 - was that what you had in mind?